### PR TITLE
conversion of tests from mockery to mockgen 

### LINKED ISSUE #104771: Automate code generated by mockery using hack/update-mocks.sh and hack/verify-mocks.sh scripts
There are a few mock files [1] in the codebase today that are generated by mockery. However, these files are not autogenerated by any scripts today. This should be fixed by phasing out `mockery` then re implement the mock files using `mockgen` and `go:generate`,  also the newly generated files using `mockgen` should be compatible with automation by using `hack/update-mocks.sh` and `hack/verify-mocks.sh` scripts which is implemented in issue https://github.com/kubernetes/kubernetes/issues/103540.

[1] https://cs.k8s.io/?q=mockery&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes
Closed PR: https://github.com/kubernetes/kubernetes/pull/103631

/kind cleanup
/sig architecture
/area code-organization
/sig testing

cc: @dims

### PR DESCRIPTION
**Pull request by vikramcse**
#### What type of PR is this?
/kind cleanup
/sig architecture
/area code-organization
/sig testing

#### What this PR does / why we need it:
conversion of tests from mockery to mockgen 

#### Which issue(s) this PR fixes:
Fixes #104771 

#### Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

**Comment by vikramcse** [comment]
Hello @dims 
I have done some progress by converting mocks from mockery(testify/mock) to gomock
May I please get ok-to-test.

Thank You 😃

File: pkg/kubelet/container/runtime_cache.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+//go:generate mockgen -source=runtime_cache.go  -destination=testing/mock_runtime_cache.go -package=testing RuntimeCache
```
[the remark this answers is not in the document]
Comment (vikramcse) [reply pkg/kubelet/container/runtime_cache.go:17]: Hello @SergeyKanzhelev 
There is a mixed pattern of naming some have `mock_foo` and others have `foo_mock`
e.g.
1. pkg/kubelet/container/testing/runtime_mock.go
2. pkg/kubelet/container/testing/mock_runtime_cache.go

Thank You :)

**Comment by xing-yang** [comment]
Changes from storage side look good.

**Pull request merged**